content/guides/reliable-testing-dataset-with-pg-dump-and-pg-restore.md

```diff
@@ -0,0 +1,290 @@
+          $POSTGRES/psql "${{ env.DEV_DATABASE_URL }}" -c "\copy public.products FROM '${{ github.workspace }}/products-subset.csv' WITH CSV HEADER"
+          $POSTGRES/psql "${{ env.DEV_DATABASE_URL }}" -c "\copy public.transactions FROM '${{ github.workspace }}/transactions-subset.csv' WITH CSV HEADER"
+```
+
+The above code snippet might look a bit complicated at first, but it’s actually not that bad—let me break it down for you.
+
+### Dump schema
+
+In this step, I use pg_dump to export the entire schema from the production database and save it to the GitHub workspace as a file named `all-schema.bak`. This file is stored in memory so it can be accessed later by the **Restore schema** step towards the end of the job.
+
+The flags used in this step are explained below:
+
+| Flag          | Meaning                                                                          |
+| ------------- | -------------------------------------------------------------------------------- |
+| -Fc           | Dumps the database in a custom format.                                           |
+| --schema-only | Dumps only the schema (table structures, indexes, constraints) without any data. |
+| -f            | Specifies the output file where the schema dump will be stored.                  |
+
+### Dump data
+
+In this step, I use psql to query the data. This is the most complex step, involving three SQL queries, each targeting one of the three tables. The queries are as follows:
+
+#### Transactions query
+
+This query selects the 50 most recent **transactions** from the **transactions** table. Depending on your requirements, you can increase the `LIMIT` or modify the query.
+
+```sql
+SELECT * FROM transactions ORDER BY transaction_id DESC LIMIT 50
+```
+
+The results are saved to the GitHub workspace memory as a file called `transactions-subset.csv`, which will be used in a later step.
+
+#### Products query
+
+This query selects **products**, but only those with a `product_id` present in the 50 most recent **transactions**:
+
+```sql
+SELECT * FROM products WHERE product_id IN (SELECT product_id FROM transactions ORDER BY transaction_id DESC LIMIT 50)
+```
+
+The results are saved to the GitHub workspace memory as a file called `products-subset.csv`, which will be used in a later step.
+
+#### Users query
+
+This query selects **users**, but only those with a `user_id` present in the 50 most recent **transactions**:
+
+```sql
+SELECT * FROM users WHERE user_id IN (SELECT user_id FROM transactions ORDER BY transaction_id DESC LIMIT 50)
+```
+
+The results are saved to the GitHub workspace memory as a file called `users-subset.csv`, which will be used in a later step.
+
+### Drop tables and schema
+
+In this step, I use psql to drop the schema and create a fresh one. Since this Action runs on a schedule, this cleanup ensures the target database is ready for new schema and data, avoiding any errors from schema changes since the last run.
```

Comment:
```suggestion
In this step, I use `psql` to drop the schema and create a fresh one. Since this Action runs on a schedule, this cleanup ensures the target database is ready for new schema and data, avoiding any errors from schema changes since the last run.
```